### PR TITLE
SLCORE-470 Change effective rule details to return all contexts

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/rules/RuleDetailsAdapter.java [hunk 1]
```diff
@@ -123,22 +121,22 @@ private static Collection<RuleDescriptionTabDto> transformSectionsButIntroductio
       if (sectionsByKey.containsKey(sectionKey)) {
         var tabContents = sectionsByKey.get(sectionKey);
         Either<RuleNonContextualSectionDto, RuleContextualSectionWithDefaultContextKeyDto> content;
-        var matchingContext = tabContents.stream().filter(c -> c.getContext().isPresent() && c.getContext().get().getKey().equals(contextKey)).findFirst();
+        var foundMatchingContext = tabContents.stream().anyMatch(c -> c.getContext().isPresent() && c.getContext().get().getKey().equals(contextKey));
```
Comment (vojtech-suchy-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/rules/RuleDetailsAdapter.java:124]: _nitpick_: we could move the calculation of `foundMatchingContext ` just before line 139 when it's needed.

**Review by vojtech-suchy-sonarsource — APPROVED** [review]
looking good :thumbsup:

File: core/src/test/java/mediumtest/EffectiveRulesMediumTests.java [hunk 2]
```diff
@@ -630,12 +516,48 @@ void it_should_split_security_hotspots_rule_description_and_adapt_title() {
...
+          .addDescriptionSections(Rules.Rule.DescriptionSection.newBuilder()
+            .setKey("introduction").setContent("intro content"))
+          .addDescriptionSections(Rules.Rule.DescriptionSection.newBuilder()
+            .setKey("how_to_fix").setContent("fix spring")
+            .setContext(Rules.Rule.DescriptionSection.Context.newBuilder().setKey("spring").setDisplayName("Spring").build()).build())
+          .addDescriptionSections(Rules.Rule.DescriptionSection.newBuilder()
+            .setKey("how_to_fix").setContent("fix struts")
+            .setContext(Rules.Rule.DescriptionSection.Context.newBuilder().setKey("struts").setDisplayName("Struts").build()).build())
+          .addDescriptionSections(Rules.Rule.DescriptionSection.newBuilder()
+            .setKey("resources").setContent("htmlContent3").build()))
```
Comment (jblievremont) [inline core/src/test/java/mediumtest/EffectiveRulesMediumTests.java:544]: Nice 👍🏻 I like that it cuts the dependency on the actual plugin.

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**